core/src/main/java/org/mule/runtime/core/api/serialization/SerializationProtocol.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.api.serialization;
+
+import org.mule.runtime.core.util.store.DeserializationPostInitialisable;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+
+/**
+ * Defines a component capable to serialize/deserialize objects into/from an array of {@link byte}s. Unlike usual serializing
+ * components, this one doesn't enforce the serialized object to implement {@link Serializable}. However, some implementations
+ * might require that condition and throw {@link IllegalArgumentException} if not met.
+ * <p/>
+ * Implementations are also responsible for the correct initialization of classes implementing the
```

Comment:
Is there a way to get rid of this restriction?. Maybe handle this internally without exposing this concept in this API?